dashboard/test/models/pd/enrollment_test.rb

```diff
@@ -490,4 +490,12 @@ class Pd::EnrollmentTest < ActiveSupport::TestCase
 
     refute teacher.permission? UserPermission::AUTHORIZED_TEACHER
   end
+
+  test 'Updating existing enrollment sets permission' do
+    teacher = create :teacher
+    workshop = create :pd_workshop, course: Pd::SharedWorkshopConstants::COURSE_CSD, num_sessions: 1
+    create :pd_enrollment, workshop: workshop, user: nil, email: teacher.email
+
+    assert teacher.permission? UserPermission::AUTHORIZED_TEACHER
+  end
```

Comment:
I expect this to fail. It needs to first update the enrollment from line 497 to associated it with the user. Ideally, first refute that the (unassociated) user is not verified.